src/main/java/com/groupon/jenkins/DockerComposeDotCi/DockerPublishPlugin.java

```diff
@@ -76,48 +76,58 @@ public boolean perform(DynamicBuild build, Launcher launcher, BuildListener list
         return true;
     }
 
-    public ShellCommands getPublishCommands(DynamicBuild build, BuildListener listener) throws IOException, InterruptedException {
+    private ShellCommands getPublishCommands(DynamicBuild build, BuildListener listener) throws IOException, InterruptedException {
+
         Map<String, Object> buildEnvironment = build.getEnvironmentWithChangeSet(listener);
         String dotCiBranch = buildEnvironment.get("DOTCI_BRANCH").toString();
         String dotCiTag = null;
         if (dotCiBranch.contains("tags/"))
             dotCiTag = buildEnvironment.get("DOTCI_TAG").toString();
-        String repoName = build.getParent().getFullName();
-        String sha = build.getSha();
+        String orgSlashRepo = build.getParent().getFullName();
+        String buildSha = build.getSha();
         int buildNumber = build.getNumber();
         boolean forcePushLatest = configuration.isForcePushLatest();
         String registryHost = configuration.getRegistryHost();
-        String dockerRegistryImageName = registryHost + repoName.toLowerCase().replaceAll("-", "_");
-        String publishProjectName = repoName.replaceAll("/", "").replaceAll("\\.", "").replaceAll("-", "").replaceAll("_", "").toLowerCase() + buildNumber;
+        String composeBuildName = orgSlashRepo.replaceAll("/", "").replaceAll("\\.", "").replaceAll("-", "")
+                .replaceAll("_", "").toLowerCase() + buildNumber;
         ShellCommands publishCommands = new ShellCommands();
 
-        // Build the image
-        LOGGER.info(String.format("Building '%s'", publishProjectName));
-        publishCommands.add(String.format("docker-compose -p %s pull", publishProjectName));
-        publishCommands.add(String.format("docker-compose -p %s build", publishProjectName));
-
-        if (dotCiTag != null) {
-            // Publish with label as tag
-            LOGGER.info(String.format("Publishing tagged image: %s:%s", dockerRegistryImageName, dotCiTag));
-            publishCommands.add(String.format("docker tag %s_%s %s:%s", publishProjectName, getComposeImageToPublish(), dockerRegistryImageName, dotCiTag));
-            publishCommands.add(String.format("docker push %s:%s", dockerRegistryImageName, dotCiTag));
+        // Build everything
+        publishCommands.add(String.format("docker-compose -p %s pull", composeBuildName));
+        publishCommands.add(String.format("docker-compose -p %s build", composeBuildName));
+
+        if (this.options instanceof Map) {
+            for (Object imageMap : ((Map) this.options).entrySet()) {
```

Comment:
You can cast this a map with types that way you don't have to do as much casting later in this block.